test/integration/xds_config_tracker_integration_test.cc

```diff
@@ -0,0 +1,248 @@
+#include <atomic>
+
+#include "envoy/config/route/v3/route.pb.h"
+#include "envoy/config/subscription.h"
+#include "envoy/config/xds_config_tracker.h"
+#include "envoy/protobuf/message_validator.h"
+#include "envoy/service/discovery/v3/discovery.pb.h"
+#include "envoy/service/runtime/v3/rtds.pb.h"
+
+#include "test/common/grpc/grpc_client_integration.h"
+#include "test/config/v2_link_hacks.h"
+#include "test/integration/http_integration.h"
+#include "test/integration/xds_config_tracker_test.pb.h"
+#include "test/test_common/registry.h"
+#include "test/test_common/utility.h"
+
+#include "absl/synchronization/mutex.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+constexpr char XDS_CLUSTER_NAME[] = "xds_cluster";
+
+/**
+ * All stats for this xds tracker. @see stats_macros.h
+ */
+#define ALL_TEST_XDS_TRACKER_STATS(COUNTER)                                                        \
+  COUNTER(on_config_accepted)                                                                      \
+  COUNTER(on_config_rejected)
+
+/**
+ * Struct definition for stats. @see stats_macros.h
+ */
+struct TestXdsTrackerStats {
+  ALL_TEST_XDS_TRACKER_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+class TestXdsConfigTracker : public Config::XdsConfigTracker {
```

Comment:
Please add comments describing what this class is and what is it used for.

Reply:
Thanks! Done.